flutter/third_party/tf-eigen.patch

```diff
@@ -0,0 +1,45 @@
+From b570eb875d9730065dcb364ef37a5ec3c885acab Mon Sep 17 00:00:00 2001
+From: "freedom\" Koan-Sin Tan" <koansin.tan@gmail.com>
+Date: Thu, 17 Mar 2022 14:51:19 +0800
+Subject: [PATCH] hacks
+
+---
+ third_party/eigen3/eigen_ios.patch | 13 +++++++++++++
+ third_party/eigen3/workspace.bzl   |  1 +
+ 2 files changed, 14 insertions(+)
+ create mode 100644 third_party/eigen3/eigen_ios.patch
+
+diff --git a/third_party/eigen3/eigen_ios.patch b/third_party/eigen3/eigen_ios.patch
+new file mode 100644
+index 00000000000..6c94319de42
+--- /dev/null
++++ b/third_party/eigen3/eigen_ios.patch
+@@ -0,0 +1,13 @@
++diff --git a/Eigen/src/Core/util/Macros.h b/Eigen/src/Core/util/Macros.h
++index e76ddd3d2..6b4fc84ec 100644
++--- a/Eigen/src/Core/util/Macros.h
+++++ b/Eigen/src/Core/util/Macros.h
++@@ -1281,7 +1281,7 @@ namespace Eigen {
++   EIGEN_MAKE_SCALAR_BINARY_OP_ONTHERIGHT(METHOD,OPNAME)
++ 
++ 
++-#if (defined(_CPPUNWIND) || defined(__EXCEPTIONS)) && !defined(EIGEN_CUDA_ARCH) && !defined(EIGEN_EXCEPTIONS) && !defined(EIGEN_USE_SYCL) && !defined(EIGEN_HIP_DEVICE_COMPILE)
+++#if (defined(_CPPUNWIND) || (defined(__EXCEPTIONS) && defined(__exceptions__))) && !defined(EIGEN_CUDA_ARCH) && !defined(EIGEN_EXCEPTIONS) && !defined(EIGEN_USE_SYCL) && !defined(EIGEN_HIP_DEVICE_COMPILE)
```

Comment:
There are troubles with exceptions because tensorflow disables them:
https://github.com/tensorflow/tensorflow/blob/6da9b31e96d5f8735821bb59109ca3e3c1a0b9ec/tensorflow/lite/build_def.bzl#L49
https://github.com/tensorflow/tensorflow/blob/ed40242d7abfb07694bd4c3061e09a958cf18c14/tensorflow/tensorflow.bzl#L380
If I disable these lines then there are no issues with exceptions in eigen files.

Are there any negative consequences for enabling exceptions that I don't see?

Reply:
Eigen is kinda last line of defense for **floating point** operations in TFLite interpreter. If a model is fully delegated to either CoreML delegate or GPU delegate, it's not used. If it's partially delegated and could be handled by XNNAPCK delegate, it's not used. I don't know the exception handling part :-)